addons/payment/controllers/portal.py

```diff
@@ -175,6 +175,19 @@ def payment_method(self, **kwargs):
         }
         return request.render('payment.payment_methods', rendering_context)
 
+    @http.route('/my/payment_method/archive', type='json', auth='user')
```

Comment:
This route is not exclusively called from the /my/payment_method page, and the user is never redirected on it either.

```suggestion
    @http.route('/payment/archive_token', type='json', auth='user')
```